ceph_iscsi_config/lun.py

```diff
@@ -433,15 +433,16 @@ def _get_next_lun_id(self, target_disks):
             lun_id_candidate += 1
         return lun_id_candidate
 
-    def map_lun(self, gateway, owner, disk):
+    def map_lun(self, gateway, owner, disk, lun_id=None):
         target_config = self.config.config['targets'][gateway.iqn]
         disk_metadata = self.config.config['disks'][disk]
         disk_metadata['owner'] = owner
         self.config.update_item("disks", disk, disk_metadata)
 
         target_disk_config = target_config['disks'].get(disk)
         if not target_disk_config:
-            lun_id = self._get_next_lun_id(target_config['disks'])
+            if lun_id is None:
+                lun_id = self._get_next_lun_id(target_config['disks'])
             target_config['disks'][disk] = {
                 'lun_id': lun_id
```

Comment:
Did you also want the lun_id to be used as the lun_id when mapped to a client, or did you also want to control that value at the client level?

You can do:

/iscsi-target...h-iscsi/disks> add rbd/testlun 100

but then if you map it to a client:

/iscsi-target...com.ceph:ini1> disk add rbd/testlun

the LUN is just the first available number:

  o- lun 0 ............................ [rbd/test1(1G), Owner: ans6

Reply:
@mikechristie looking into how `lrbd` works I see that the mapped lun also uses the specified lun_id:

![Screenshot from 2019-10-21 12-24-16](https://user-images.githubusercontent.com/14297426/67202103-c8a55800-f3ff-11e9-83ff-97cd9a5c6574.png)

Thanks for the hint, I'll adapt this PR.

Reply:
@mikechristie this should be fixed now.
Clients and groups will now read the `lun_id` that is already defined at target disk level (instead of generating a new one).